File: packages/grafana-faro/android/src/main/java/io/capawesome/capacitorjs/plugins/grafanafaro/classes/options/AppMetadata.java
package io.capawesome.capacitorjs.plugins.grafanafaro.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.grafanafaro.GrafanaFaroPlugin;

public class AppMetadata {

    @Nullable
    private final String environment;

    @NonNull
    private final String name;

    @Nullable
    private final String namespace;

    @Nullable
    private final String version;

    public AppMetadata(@NonNull JSObject source) throws Exception {
        String name = source.getString("name");
        if (name == null || name.isEmpty()) {
            throw new Exception(GrafanaFaroPlugin.ERROR_APP_NAME_MISSING);
        }
        this.environment = source.getString("environment");
        this.name = name;
        this.namespace = source.getString("namespace");
        this.version = source.getString("version");
    }

    @Nullable
    public String getEnvironment() {
        return environment;
    }

    @NonNull
    public String getName() {
        return name;
    }

    @Nullable
    public String getNamespace() {
        return namespace;
    }

    @Nullable
    public String getVersion() {
        return version;
    }
}

File: packages/grafana-faro/ios/Plugin/Classes/Options/AppMetadata.swift
import Capacitor
import Foundation

@objc public class AppMetadata: NSObject {
    private let environment: String?
    private let name: String
    private let namespace: String?
    private let version: String?

    init(source: JSObject) throws {
        guard let name = source["name"] as? String, !name.isEmpty else {
            throw CustomError.appNameMissing
        }
        self.environment = source["environment"] as? String
        self.name = name
        self.namespace = source["namespace"] as? String
        self.version = source["version"] as? String
    }

    func getEnvironment() -> String? {
        return environment
    }

    func getName() -> String {
        return name
    }

    func getNamespace() -> String? {
        return namespace
    }

    func getVersion() -> String? {
        return version
    }
}
